cypress/utils/apps/query_enhancements/inspect.js

```diff
@@ -0,0 +1,250 @@
+  const seconds = date.getSeconds().toString().padStart(2, '0');
+  const milliseconds = date.getMilliseconds().toString().padStart(3, '0');
+
+  return `${month} ${day}, ${year} @ ${hours}:${minutes}:${seconds}.${milliseconds}`;
+}
+
+/**
+ * Format value to the corresponding string expected in the UI.
+ * Formats commas in numbers, and dates.
+ * @param {string} value - field value to be formatted.
+ * @returns {string}
+ */
+const formatValue = (value) => {
+  switch (typeof value) {
+    case 'number':
+      return value.toLocaleString();
+    case 'string':
+      // SQL and PPL date string uses format YYYY-MM-DD HH:MM:SS.SSS, we want to convert to ISO date.
+      const potentialDate = new Date(value.replace(' ', 'T'));
+      if (potentialDate instanceof Date && !isNaN(potentialDate)) {
+        return formatDate(potentialDate);
+      }
+    default:
+      return value;
+  }
+};
+
+/**
+ * Returns the SavedSearchTestConfig for the provided dataset, datasetType, and language
+ * @param {string} dataset - the dataset name
+ * @param {QueryEnhancementDataset} datasetType - the type of the dataset
+ * @param {QueryEnhancementLanguageData} language - the relevant data for the query language to use
+ * @returns {InspectTestConfig}
+ */
+export const generateInspectTestConfiguration = (dataset, datasetType, language) => {
+  return {
+    dataset,
+    datasetType,
+    language: language.name,
+    testName: `dataset: ${datasetType} and language: ${language.name}`,
+  };
+};
+
+/**
+ * Flatten the object in the format e.g. {
+                        "FlightNum": "SYS76NQ",
+                        "DistanceKilometers": 7030.318886915258,
+                        "timestamp": "2025-01-27T16:06:01",
+                        "DestLocation": {
+                            "lat": "38.94449997",
+                            "lon": "-77.45580292"
+                        },
+                    } and format the values.
+ * @param {object} obj - Object to be flattened.
+ * @param {string?} parentKey - Starting prefix which will be prepended to all properties. Leave blank if not needed.
+ * @param {object?} properties - Object to be modified.
+ * @returns {object}
+ */
+
+const flattenObjectAndFormatValues = (obj, parentKey, properties = {}) => {
+  for (const [key, value] of Object.entries(obj)) {
+    const property = parentKey ? `${parentKey}.${key}` : key;
+    if (Array.isArray(value)) {
+      // SQL and PPL arrays put objects into their own arrays with no other objects in the array.
+      flattenObjectAndFormatValues(value[0], property, properties);
+    } else if (typeof value === 'object' && value !== null) {
+      // Assume that the value will never be a function.
+      flattenObjectAndFormatValues(value, property, properties);
+    } else {
+      properties[property] = formatValue(value);
+    }
+  }
+  return properties;
+};
+
+/**
+ * Flatten the array in the format e.g. [
```

Comment:
same issue

Reply:
description is not clear. suggest to change to `Flattens an array of field definitions and extracts values for a specific row`